verticapy/stats/tools.py

```diff
@@ -1424,3 +1425,124 @@ def variance_inflation_factor(vdf: vDataFrame, X: list, X_idx: int = None):
         raise ParameterError(
             f"Wrong type for Parameter X_idx.\nExpected integer, found {type(X_idx)}."
         )
+
+# ---#
+def is_dir_path(adj_list: dict, X:str, Y:str):
+    """
+---------------------------------------------------------------------------
+Determines if there is a directed path from X -> Y given an adjacency list.
+
+Parameters
+----------
+adj_list: dict
+    Adjacency List where the values are the other keys
```

Comment:
It is nice that this function is added as a generic tool that might be used in the future somewhere out of the scope of this PR. Nevertheless, it would be nice for the same reason to make its functionality as clear as possible.
For example, is it a correct assumption that the `adj_list` represents a generic directed graph (with potential cycles)?

Reply:
It is correct to assume it represents a generic directed graph. The values of the dictionary can either be a set or list. While I do think this could work with a list[list] or list[set], I think it's best to keep the adj_list struct as a dictionary of lists or sets because it allows the function to have nodes be related unique keys.